listapp-react/src/App.js

```diff
@@ -0,0 +1,174 @@
+import './App.css';
+import LineList from './LineList.js';
+import {useState} from 'react';
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+
+function App() {
+
+    // Each list item is initialized with the fields shown below
+    const initialData = [
+        {
+            key: 0,
+            text: "Tap to Add Note",
+            checked: false,
+            check_visible: false,
+            text_visible: true,
+            select_visible: false
+        }
+    ]
+
+    // state data to be used later
+    const [list, setList] = useState(initialData);
+    const [selected, setSelected] = useState([]);
+    const [edited, setEdited] = useState(-1);
+    const [hidden, setHidden] = useState(false);
+    const [showWarning, setWarning] = useState(false);
+
+    // stores data on what to display
+    let showHideButton = (list.filter((item) => item.checked === true)).length > 0;
+    let showDeleteButton = selected.length > 0;
+    let disableChecks = (edited !== -1);
+
+    // update the edited state with the line key if we've currently clicked onto a line, -1 otherwise
+    function handleLineEdited(lineID) {
+
+        // if we're editing the bottommost 'Tap to Add Note' line and the text has changed, update our data/state
+        if (edited === list[list.length-1].key && list[list.length-1].text !== "Tap to Add Note") {
+            if (list[list.length-1].text === "") {
+                handleItemChanged(list[list.length-1].key, "text", "Tap to Add Note");
+
+            } else {
+                // display check and select box for added note, create tap line
+                list[list.length-1].check_visible = true;
+                list[list.length-1].select_visible = true;
+                handleItemAdded("Tap to Add Note");
+
+            }
+        }
+        if (lineID === list[list.length-1].key && edited !== lineID) {
+            handleItemChanged(lineID, "text", "")
+        }
+
+        setEdited(lineID) // update edited line state
+    }
+
+    // changes line data for textboxes, checkboxes, or special key presses
+    function handleItemChanged(itemID, field, newValue) {
+
+        if (field === "text") {
+            return (
+                setList(list.map(
+                    p => p.key === itemID ? {...p, [field]:newValue} : p))
+            );
+        }
+        if (field === "checkbox") {
```

Comment:
If you're getting in a newValue passed in, why not just use your code in 59-62 and ignore checking the field names?